client/transactions/filters/index.tsx

```diff
@@ -8,45 +8,46 @@ import { getQuery } from '@woocommerce/navigation';
 /**
  * Internal dependencies
  */
-import { filters, advancedFilters, TransactionsFilterType } from './config';
+import { getFilters, getAdvancedFilters } from './config';
 import { formatCurrencyName } from '../../utils/currency';
 import './style.scss';
+import { __ } from '@wordpress/i18n/build-types';
 
 interface TransactionsFiltersProps {
 	storeCurrencies?: string[];
+	customerCurrencies?: string[];
 }
 
 export const TransactionsFilters = ( {
 	storeCurrencies,
+	customerCurrencies,
 }: TransactionsFiltersProps ): JSX.Element => {
-	const populateDepositCurrencies = (
-		filtersConfiguration: TransactionsFilterType[]
-	) => {
-		filtersConfiguration.forEach( ( filter ) => {
-			if ( 'store_currency_is' === filter.param ) {
-				const currencies = storeCurrencies || [];
-				// Generate select options: pick the first one (default) and add provided currencies
-				filter.filters = [
-					filter.filters[ 0 ],
-					...currencies.map( ( currencyCode: string ) => ( {
-						label: formatCurrencyName( currencyCode ),
-						value: currencyCode,
-					} ) ),
-				];
-				// Show the select when several currencies are available.
-				if ( 2 < filter.filters.length ) {
-					filter.showFilters = () => true;
-				}
-			}
-		} );
-		return filtersConfiguration;
-	};
+	const customerCurrencyOptions = customerCurrencies?.map(
+		( currencyCode: string ) => ( {
+			label: formatCurrencyName( currencyCode ),
+			value: currencyCode,
+		} )
+	);
```

Comment:
This is a good example where `useMemo` could be used to avoid doing complex operations on every render. It's not that heavy, but we are mapping the array and calling the get filter method to generate it.

We can store it as a constant using a code similar to the following one, and the same will apply to `filters` with `storeCurrencies`.

Be aware, that we should evaluate when `useMemo` should be used, as we can end having a performance hit while trying to improve it, by just using `useMemo` or similar hooks everywhere.

```js
const advancedFilters = useMemo(
    () =>
        getAdvancedFilters(
            customerCurrencies.map( ( currencyCode: string ) => ( {
                label: formatCurrencyName( currencyCode ),
                value: currencyCode,
            } ) )
        ),
    [ customerCurrencies ]
);
```

Reply:
Thanks, I tested this out and it seems to improve the performance (at least from a UI POV) because it means it doesn't reload the list when just choosing a currency on the customer currency filter and clicking filter.

Updated in 453d71d5c510c20e4a47881f4d23fa62c2fb0683.